recipes/c4core/all/conanfile.py

```diff
@@ -42,49 +45,53 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def requirements(self):
-        self.requires("fast_float/3.4.0")
+        if self.options.with_fast_float:
+            self.requires("fast_float/3.5.1")
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "11")
+            check_min_cppstd(self, "11")
 
-        ## clang with libc++ is not supported. It is already fixed at 2022-01-03.
-        if tools.Version(self.version) <= "0.1.8":
-            if (self.settings.compiler == "clang" and self.settings.compiler.get_safe("libcxx") == "libc++"):
-                raise ConanInvalidConfiguration(
-                    "{}/{} doesn't support clang with libc++".format(self.name, self.version),
-                )
+        ## clang with libc++ is not supported. It is already fixed since 0.1.9.
+        if Version(self.version) <= "0.1.8":
+            if self.settings.compiler in ["clang", "apple-clang"] and \
+                self.settings.compiler.get_safe("libcxx") == "libc++":
+                raise ConanInvalidConfiguration(f"{self.ref} doesn't support clang with libc++")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["C4CORE_WITH_FASTFLOAT"] = bool(self.options.with_fast_float)
```

Comment:
cache variables should be used sparingly, those go at the command line which is great for override policy and such


```suggestion
        tc.variables["C4CORE_WITH_FASTFLOAT"] = bool(self.options.with_fast_float)
```

Reply:
@prince-chrismc 
Thank you for your comment.
I now understand the difference between cache_variables and variables.